src/jit/lsra.cpp

```diff
@@ -7005,23 +7005,34 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock, bool alloc
                 // Is there another interval currently assigned to this register?  If so unassign it.
                 if (assignedInterval != nullptr)
                 {
+#ifdef _TARGET_ARM_
+                    if (assignedInterval->assignedReg == targetRegRecord ||
+                        isSecondHalfReg(targetRegRecord, assignedInterval))
+#else
                     if (assignedInterval->assignedReg == targetRegRecord)
+#endif
```

Comment:
Here you could define a method such as `isAssignedToInterval(Interval, RegRecord)` that could do the simple equivalence check when not `_TARGET_ARM_`.